src/dmd/link.d

```diff
@@ -1003,30 +1025,39 @@ version (Windows)
          */
         const(char)* linkerPath(bool x64)
         {
-            if (auto p = getVCDir(VCDir.Bin, false)) // prefer 32-bit linker in case of cross-compilation
+            const(char)* addpath;
+            if (auto p = getVCBinDir(x64, addpath))
             {
                 OutBuffer cmdbuf;
                 cmdbuf.writestring(p);
                 cmdbuf.writestring(r"\link.exe");
-                if (VSInstallDir)
+                if (addpath)
                 {
                     // debug info needs DLLs from $(VSInstallDir)\Common7\IDE for most linker versions
                     //  so prepend it too the PATH environment variable
                     const char* path = getenv("PATH");
-                    const char* idepath = FileName.combine(VSInstallDir, r"Common7\IDE");
-                    auto pathlen = strlen(path);
-                    auto idepathlen = strlen(idepath);
+                    const pathlen = strlen(path);
+                    const addpathlen = strlen(addpath);
 
-                    char* npath = cast(char*)mem.xmalloc(5 + pathlen + 1 + idepathlen + 1);
+                    char* npath = cast(char*)mem.xmalloc(5 + pathlen + 1 + addpathlen + 1);
                     memcpy(npath, "PATH=".ptr, 5);
-                    memcpy(npath + 5, idepath, idepathlen);
-                    npath[5 + idepathlen] = ';';
-                    memcpy(npath + 5 + idepathlen + 1, path, pathlen + 1);
+                    memcpy(npath + 5, addpath, addpathlen);
+                    npath[5 + addpathlen] = ';';
+                    memcpy(npath + 5 + addpathlen + 1, path, pathlen + 1);
                     putenv(npath);
                 }
                 return cmdbuf.extractString();
             }
 
+            // try lld-link.exe alongside dmd.exe
+            char[MAX_PATH + 1] dmdpath = void;
+            if (GetModuleFileNameA(null, dmdpath.ptr, dmdpath.length) <= MAX_PATH)
+            {
+                auto lldpath = FileName.replaceName(dmdpath.ptr, "lld-link.exe");
```

Comment:
Likely will break if we shipped a 32-bit bin/lld.exe with a 64-bit bin64/dmd.exe.

Reply:
We might also add a 64-bit lld.exe in that case.